pkg/util/timeutil/time_source.go

```diff
@@ -54,9 +54,14 @@ type TickerI interface {
 	Ch() <-chan time.Time
 }
 
-// DefaultTimeSource is a TimeSource using the system clock.
+// DefaultTimeSource is a TimeSource using the system clock. Use
+// SystemTimeSource for a singleton.
 type DefaultTimeSource struct{}
 
+// SystemTimeSource is an instance of DefaultTimeSource that can be used as a
+// singleton.
+var SystemTimeSource = DefaultTimeSource{}
```

Comment:
why? empty structs have a fun property that they always have the same address: https://go.dev/play/p/qe2WDZ-lXBI